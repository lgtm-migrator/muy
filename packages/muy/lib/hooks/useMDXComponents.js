import Blockquote from "../Blockquote"
import Divider from "@material-ui/core/Divider"
import { H1, H2, H3, H4, H5, H6 } from "../Heading"
import Image from "../Image"
import Link from "@material-ui/core/Link"
import List from "@material-ui/core/List"
import ListItemWithText from "../ListItemWithText"
import Paragraph from "../Paragraph"
import Table from "@material-ui/core/Table"
import TableCell from "@material-ui/core/TableCell"
import TableRow from "@material-ui/core/TableRow"
import React from "react"

const useMDXComponents = () => ({
  a: Link,
  blockquote: Blockquote,
  h1: (props) => <H1 gutterBottom={true} {...props} />,
  h2: (props) => <H2 gutterBottom={true} {...props} />,
  h3: (props) => <H3 gutterBottom={true} {...props} />,
  h4: (props) => <H4 gutterBottom={true} {...props} />,
  h5: (props) => <H5 gutterBottom={true} {...props} />,
  h6: (props) => <H6 gutterBottom={true} {...props} />,
  hr: Divider,
  img: Image,
  li: ListItemWithText,
  p: Paragraph,
  table: Table,
  td: TableCell,
  tr: TableRow,
  ul: List,
})

export default useMDXComponents
