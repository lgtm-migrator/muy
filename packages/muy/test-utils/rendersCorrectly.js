import renderer from "react-test-renderer"

const rendersCorrectly = component => {
  const tree = renderer.create(component).toJSON()
  return expect(tree).toMatchSnapshot()
}

export default rendersCorrectly
