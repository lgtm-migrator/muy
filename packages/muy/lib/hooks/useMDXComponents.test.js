import React from "react"
import useMDXComponents from "./useMDXComponents"
import rendersCorrectly from "../../test-utils/rendersCorrectly"

describe("Use MDX components hook", () => {
  let components

  beforeAll(() => {
    components = useMDXComponents()
  })

  it("should return an object", () => {
    expect(components).toBeInstanceOf(Object)
  })

  const expectedProperties = [
    "blockquote",
    "h1",
    "h2",
    "h3",
    "h4",
    "h5",
    "h6",
    "hr",
    "img",
    "li",
    "p",
    "td",
    "tr",
    "ul",
  ]

  const expectedPropertiesWithChildren = ["a", "table"]

  describe.each(expectedProperties)("component for %p", (property) => {
    it("should render correctly without props", () => {
      const Component = components[property]
      rendersCorrectly(<Component />)
    })
  })

  describe.each(expectedPropertiesWithChildren)(
    "component for %p",
    (property) => {
      it("should render correctly with children", () => {
        const Component = components[property]
        rendersCorrectly(<Component>Test children</Component>)
      })
    }
  )
})
