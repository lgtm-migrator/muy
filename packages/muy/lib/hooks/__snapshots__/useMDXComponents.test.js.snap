// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Use MDX components hook component for "a" should render correctly with children 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  onBlur={[Function]}
  onFocus={[Function]}
>
  Test children
</a>
`;

exports[`Use MDX components hook component for "blockquote" should render correctly without props 1`] = `
<blockquote
  className="MuiBox-root MuiBox-root-3 makeStyles-root-1"
/>
`;

exports[`Use MDX components hook component for "h1" should render correctly without props 1`] = `
<h1
  className="MuiTypography-root MuiTypography-h1 MuiTypography-gutterBottom"
/>
`;

exports[`Use MDX components hook component for "h2" should render correctly without props 1`] = `
<h2
  className="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom"
/>
`;

exports[`Use MDX components hook component for "h3" should render correctly without props 1`] = `
<h3
  className="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom"
/>
`;

exports[`Use MDX components hook component for "h4" should render correctly without props 1`] = `
<h4
  className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
/>
`;

exports[`Use MDX components hook component for "h5" should render correctly without props 1`] = `
<h5
  className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
/>
`;

exports[`Use MDX components hook component for "h6" should render correctly without props 1`] = `
<h6
  className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
/>
`;

exports[`Use MDX components hook component for "hr" should render correctly without props 1`] = `
<hr
  className="MuiDivider-root"
/>
`;

exports[`Use MDX components hook component for "img" should render correctly without props 1`] = `
<img
  className="makeStyles-root-41"
/>
`;

exports[`Use MDX components hook component for "li" should render correctly without props 1`] = `
<li
  className="MuiListItem-root MuiListItem-gutters"
  disabled={false}
>
  <div
    className="MuiListItemText-root"
  />
</li>
`;

exports[`Use MDX components hook component for "p" should render correctly without props 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
/>
`;

exports[`Use MDX components hook component for "table" should render correctly with children 1`] = `
<table
  className="MuiTable-root"
  role={null}
>
  Test children
</table>
`;

exports[`Use MDX components hook component for "td" should render correctly without props 1`] = `
<td
  aria-sort={null}
  className="MuiTableCell-root"
/>
`;

exports[`Use MDX components hook component for "tr" should render correctly without props 1`] = `
<tr
  className="MuiTableRow-root"
  role={null}
/>
`;

exports[`Use MDX components hook component for "ul" should render correctly without props 1`] = `
<ul
  className="MuiList-root MuiList-padding"
/>
`;
